### PR TITLE
 Add overlay UI element

### PR DESCRIPTION
**Pull request by g547315**
## References
[#15090](https://github.com/jupyterlab/jupyterlab/issues/15090)
## Code changes
Added and styled Add overlay UI element

## User-facing changes

- A UI element that will be seen when functionality is trigger (pressing Alt or Alt + Shift)
- To see the full interned functionality please combine this PR with:
 extend keystrokes to mod keys:https://github.com/jupyterlab/lumino/pull/637

## Backwards-incompatible changes

None

### CODE REVIEW

**Comment by krassowski** [comment]
Before putting more work into this, it might benefit from some more discussion on how we want to implement this. I can imagine that we might in future want to add the shortcut overlays to other UI elements too. Maybe a more generic approach would be better? I can imagine using a data attribute on elements with shortcuts to be highlighted and then creating the overlays as needed. In particular this could also work for toolbar buttons without duplicating the code. As for the trigger (when the overlay is shown), possibly it could be a part of lumino keybindings processing loop?

CC @brichet who is working on rewriting toolbar buttons and also looked into keybindings recently to ask about thoughts on the implementation proposed above.

**Comment by brichet** [comment]
Thanks @krassowski for the ping.
I agree that we should use a more generic approach.
Using a data attribute on elements could be a good way of displaying the overlay on any element. This would work for toolbar buttons and should also work for extension elements.

AFAIK there is currently no link between an 'active' element (clickable) and its shortcut equivalent. For toolbar buttons the shortcuts could probably be inferred from the command (which are identical). However I don't know how it could be generalized to any element, like the tabbar, which is not described the same way...

File: .readthedocs.yaml [hunk 1]
```diff
@@ -1,5 +1,10 @@
 version: 2
-sphinx:
-  configuration: docs/source/conf.py
+
+build:
+  os: ubuntu-22.04
```
Comment (bollwyvl) [inline .readthedocs.yaml:4]: just needed to backfill this to get a preview build :heart: 
- build https://readthedocs.org/projects/lumino/builds/22062754/
- preview https://lumino--633.org.readthedocs.build/en/633/examples/dockpanel/index.html

**Comment by tonyfast** [comment]
i am trying to test this pull request on https://lumino--633.org.readthedocs.build/en/633/examples/dockpanel/index.html . i am not seeing anything happen when i press the prescribed key combinations. is there anyway to update to the examples with this working feature so its easier to test?

**Comment by g547315** [comment]
> i am trying to test this pull request on https://lumino--633.org.readthedocs.build/en/633/examples/dockpanel/index.html . i am not seeing anything happen when i press the prescribed key combinations. is there anyway to update to the examples with this working feature so its easier to test?

This by itself will just add a hidden div element called lm-TabBar-UI-Overlay to every tab on the side panels 

- To see the interned functionality (without the short-cut number in the overlay) please combine this PR with:
 extend keystrokes to mod keys:https://github.com/jupyterlab/lumino/pull/637

**Comment by tonyfast** [comment]
>     * To see the interned functionality (without the short-cut number in the overlay) please combine this PR with:
>       extend keystrokes to mod keys:[extend keystrokes to mod keys #637](https://github.com/jupyterlab/lumino/pull/637)

i do not have an easy way to test accessibility builds on multiple branches. it is really time consuming and technically challenging to build these results then test them with assistve tech. i'd really like to help move these pull requests forward, but their structure makes it really hard to test. is there anyway to combine these pull requests into something that can be tested easier? ideally testing the outcomes on the lumino examples would be best.

File: packages/widgets/src/tabbar.ts [hunk 2]
```diff
@@ -1755,6 +1756,14 @@ export namespace TabBar {
       return h.div({ className: 'lm-TabBar-tabCloseIcon' });
     }
 
+    renderOverlay(data: IRenderData<any>): VirtualElement {
+      if (data.title.label == 'Launcher') {
```
[the remark this answers is not in the document]
Comment (m158261) [reply packages/widgets/src/tabbar.ts]: Hey @krassowski, I have updated this to use dataset.id which is not dependent on internationalisation.

**Pull request closed without merging**